### PR TITLE
Display authors in personal assistant view

### LINKED ISSUE #2912: [Ahuna] Display an assistant's `authors`
Related [discussion](https://dust4ai.slack.com/archives/C050A0S2Z7F/p1702896396237949)

Authors are all users who edited an assistant. We display only "Major Contributors": the last 3 editors (ordered by edit date)
Display occurs:
- in assistant gallery
- in "My assistants" / "Workspace assistants"
- in an assistant Detailed View

Figma  [here](https://www.figma.com/file/QOxHwppG64GtPocpDhS6Y7/Product?type=design&node-id=2914-9314&mode=design&t=5iTBhOnZdcXtMyKV-4) to be updated accordingly

Detailed assistant view should also show last edit time

### PR DESCRIPTION
**Pull request by flvndvd**
Slack context: [thread](https://dust4ai.slack.com/archives/C050A0S2Z7F/p1702910140159249).
Figma: [here](https://www.figma.com/file/QOxHwppG64GtPocpDhS6Y7/Product?node-id=2914%3A9312&mode=dev).

This PR resolves https://github.com/dust-tt/dust/issues/2912 by implementing a feature to display the last 3 authors in the personal gallery. Due to Sequelize's limitations in not fully exposing PostgreSQL's capabilities, an efficient MySQL query at runtime wasn't feasible without potentially impacting endpoint performance. A raw query was considered. However, this would bypass Sequelize's relational abstractions.

The chosen approach involves using a Redis sorted set as a "cached view" to store the most recent author ids for each assistant, with a TTL of 3 days. This leverages Redis's automatic data lifecycle management, allowing us to append new data without manual cleanup of older entries. The sorted set is refreshed with every assistant update.

**Key points:**
- Implements Redis caching for recent author data.
- API side now fetches member names, a temporary pattern until the need for broader member awareness arises on the client side.
- Introduces `withAuthors` query parameter to `/agent_configurations` endpoint, similar to `withUsage`. This will be restructured in a future PR for a more refined selection mechanism.

image

### CODE REVIEW

File: front/pages/api/w/[wId]/assistant/agent_configurations/index.ts [hunk 1]
```diff
@@ -115,6 +118,29 @@ async function handler(
           );
         });
       }
+
+      if (withAuthors === "true") {
+        const members = await getMembers(auth);
```
Comment (spolu) [inline front/pages/api/w/[wId]/assistant/agent_configurations/index.ts:123]: This one is pretty big and should be run cached as well no?

**Review by spolu — COMMENTED** [review]
Looks great! Left a few comments to discuss :pray:

File: front/pages/api/w/[wId]/assistant/agent_configurations/index.ts [hunk 2]
```diff
@@ -115,6 +118,29 @@ async function handler(
...
+
+        agentConfigurations = await Promise.all(
+          agentConfigurations.map(
+            async (agentConfiguration): Promise<AgentConfigurationType> => {
+              return {
+                ...agentConfiguration,
+                lastAuthors: await getAgentRecentAuthors(
+                  {
+                    agentConfiguration,
+                    workspaceId: owner.sId,
```
Comment (spolu) [inline front/pages/api/w/[wId]/assistant/agent_configurations/index.ts:133]: As per IRL let's push auth as far as possible

Comment (flvndvd) [reply front/pages/api/w/[wId]/assistant/agent_configurations/index.ts]: Fixed.

File: front/lib/api/assistant/recent_authors.ts [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+  if (recentAuthorIds.length === 0) {
+    // Populate from the database and store in Redis if the entry is not already present.
+    recentAuthorIds = await populateAuthorIdsFromDb(agentId, workspaceId);
+  }
+
+  // Consider moving this logic to the FE if we need to fetch members in different places.
+  return renderAuthors(recentAuthorIds, members, currentUserId);
+}
+
+export async function agentConfigurationWasUpdatedBy({
```
Comment (spolu) [inline front/lib/api/assistant/recent_authors.ts:155]: let's take `auth: Authenticator` and `agent: AgentCongifurationType` here to make things more "typed"

**Review by spolu — APPROVED** [review]
Left another request to use Authenthicator/AgentConfigurationType instead of strings.

Feel free to merge once addressed :+1: 

cc @lasryaric for redis related stuff

LGTM :100:

File: front/pages/api/w/[wId]/assistant/agent_configurations/index.ts [hunk 4]
```diff
@@ -115,6 +118,28 @@ async function handler(
           );
         });
       }
+
+      if (withAuthors === "true") {
+        const members = await getMembers(auth);
+
+        agentConfigurations = await Promise.all(
+          agentConfigurations.map(
+            async (agentConfiguration): Promise<AgentConfigurationType> => {
```
Comment (spolu) [inline front/pages/api/w/[wId]/assistant/agent_configurations/index.ts:127]: feel free to use agent in place of agentConfiguration

File: front/lib/api/assistant/recent_authors.ts [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+        return members.find((m) => m.id === authorId)?.fullName ?? null;
+      })
+      // Filter out `null` authors.
+      .filter((name): name is string => name !== null)
+  );
+}
+
+export async function getAgentRecentAuthors(
+  {
+    agentConfiguration,
```
Comment (spolu) [inline front/lib/api/assistant/recent_authors.ts:114]: feel free to call it agent if you want to save a few characters

**Pull request merged**